source/common/upstream/resource_manager_impl.h

```diff
@@ -40,23 +43,35 @@ class ResourceManagerImpl : public ResourceManager {
 
 private:
   struct ResourceImpl : public Resource {
-    ResourceImpl(uint64_t max, Runtime::Loader& runtime, const std::string& runtime_key)
-        : max_(max), runtime_(runtime), runtime_key_(runtime_key) {}
+    ResourceImpl(uint64_t max, Runtime::Loader& runtime, const std::string& runtime_key,
+                 Stats::Gauge& open_gauge)
+        : max_(max), runtime_(runtime), runtime_key_(runtime_key), open_gauge_(open_gauge) {}
     ~ResourceImpl() { ASSERT(current_ == 0); }
 
     // Upstream::Resource
     bool canCreate() override { return current_ < max(); }
-    void inc() override { current_++; }
+    void inc() override {
+      current_++;
+      open_gauge_.set(canCreate() ? 0 : 1);
```

Comment:
the only problem I see here is that I don't think this stat is going to work properly if you have both default and high priority circuit breakers. It might bounce back and forth.

Given that we now have people complaining about too many stats (@alyssawilk just brought this up on another PR), I'm wondering if we should do two things:
1) Make this a configured option.
2) Dynamically create a new stat tree with the cluster tree, and properly handle default and high if it exists? So maybe something like `clusters.foo.circuit_breakers.default.cx_open` or something like that?

Reply:
@mattklein123 do we want every stats provider to be adding their own configuration options? Or should we have a common path/scope method for this? I thought @ambuc was adding something like this.

Reply:
https://github.com/envoyproxy/envoy/pull/4498 - why are "too many stats" still an issue?

Reply:
I've created the sub-tree in 32413d0.

Reply:
Yeah I guess if we can remove stats w/ config we don't need a config option, however having these stats in a dedicated tree makes them easy to remove.